R/write.magpie.R

```diff
@@ -172,17 +172,28 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
         stop("The package \"raster\" is required!")
       }
       format <- c(asc = "ascii", grd = "raster", tif = "GTiff")
-      if (ndata(x) != 1) stop("Currently no support for multiple variables for file type \"", file_type,
-                              "\". Please store each variable separately.")
-      rx <- as.RasterBrick(x)
+      if (is.magpie(x)) {
+        if (ndata(x) != 1) stop("Currently no support for multiple variables for file type \"", file_type,
+                                "\". Please store each variable separately.")
+        varname <- getItems(x, dim = 3)
+        zunit <- ifelse(all(isYear(getYears(x))), "years", "")
+        x <- as.RasterBrick(x)
+      } else if (class(x) == "RasterBrick") {
+        tmp <- names(x)
+        tmp <- strsplit(tmp,"\\..")
+        years <- sort(unique(unlist(lapply(tmp,function(x) x[1]))))
+        varname <- sort(unique(unlist(lapply(tmp,function(x) x[2]))))
+        zunit <- ifelse(all(isYear(years)), "years", "")
+        years <- as.numeric(gsub("y","",years))
```

Comment:
is the `years` variable being used anywhere afterwards?

Reply:
No, it's not needed. Deleted.